### ISSUE TITLE
custom game ACCEPT button

### ISSUE THREAD

**Issue by elpaxdota** [Arcana/node-dota2#444]
I'm used this library for creating custom game lobby. When I'm launching game\lobby popups default green ACCEPT button and bot can't accept it ( because it's bot :) ) and I can't find any method to do that. So my questions: 1. Can I somehow spectate custom game (as normal game) ? 2. How can I press ACCEPT button with bot ?

**Comment by Crazy-Duck** [comment]
Unfortunately that is not possible (nor is it trivial to add) at this time. There are two different communication protocols the Dota2 client uses to communicate with Valve servers. The first one, which is implemented in this library, is Google protocol buffers, which are used to communicate with the Dota 2 Game Coordinator. The Game Coordinator handles all general stuff, inventory management, chat, creating lobbies, searching games, ... however it does not handle the game itself. This is where the second communication protocol comes in, i.e. Steam datagram packets. Once a game has been found, the GC communicates the relay coordinates to all participants. These are used to establish the communication with the actual game server that handles the match. This whole second part is not implemented in this library and is currently considered out of scope.

**Comment by rossengeorgiev** [comment]
It's pointless having your bot accept as it can't connect. In the case of customs games, that will also cause the game to never start. So whatever you are trying to do is currently not possible with custom games.

**Comment by SmaugTheGreat** [comment]
The ReadUp message requires an encrypted key in order to accept your message. Quite obviously, this is to prevent bots from auto accepting matches. Releasing this functionality in a public respository like this one would be pointless, as Valve would simply keep changing the encryption algorithm.

**Comment by c00kie17** [comment]
@lemeshovich **WHY** are you going on every single thread and requesting an answer to this question. 
Anyway let me make it simple for you and anyone else who is planning to do this. 
You can find out how to interact with the servers on the [Steam Networking](https://partner.steamgames.com/doc/features/multiplayer/networking) page. 

### How would someone go about doing this. (why this is not possible):
* To interact with the steam SDK servers you need a `key` which of-course Valve wont just hand to you  
* Lets say you figure this key out every new build of dota2 the key will change so you'll have to go over figuring out how to find the key again 
* You have somehow cracked the key problem, The next problem you face is that the steamworks SDK will reject a connection to the game server if it does not find a graphics interface which responds back with its state, so for every bot you create you need a PC with a monitor and at that point I think that what we are trying to achieve with this module here is just not worth it.

### A more realistic approach:
* let someone in the lobby start it after your bot leaves.
* Valve has not properly implemented a spectator slot for custom games once this is done I think it will be possible to achieve what you are asking
